math/src/elliptic_curve/short_weierstrass/curves/bls12_381/field_extension.rs

```diff
@@ -282,6 +283,102 @@ impl FieldElement<Degree12ExtensionField> {
             ]),
         ])
     }
+
+    // Follows https://www.iacr.org/archive/pkc2010/60560212/60560212.pdf section 3.2
+    pub fn cyclotomic_square(&self) -> Self {
+        // f² = A' + B'⍵² + 2⍵*√A * √B , where A' = A₀ + A₁y + A₂y² , B' = B₀ + B₁y + B₂y² and x², y = ⍵²
```

Comment:
You need to document clearly whether you use quadratic over cubic (Fp2->Fp6->Fp12) or cubic over quadratic like in the paper (Fp2->Fp4->Fp12)

Reply:
> You need to document clearly whether you use quadratic over cubic (Fp2->Fp6->Fp12) or cubic over quadratic like in the paper (Fp2->Fp4->Fp12)

Yeah that's a good point. Though I felt mentioning the section in the paper would be self explanatory.